dogen/cli.py

```diff
@@ -64,9 +65,9 @@ def parse_args(self):
 
     def run(self):
 
-        tools.artifact_fetcher = partial(tools.fetch_artifact,
-                                         directory=self.args.target,
-                                         ssl_verify=not self.args.skip_ssl_verification)
+        tools.Artifact.ssl_verify = not self.args.skip_ssl_verification
```

Comment:
You set class variables in execution? That looks a bit weird, I would probably set it to a particular instance of the class.